### PR TITLE
Some clarifications and style fixes in E2EE-related areas

### PR DESCRIPTION
**Pull request by dkasak**
Changes include:

- Link to correct Megolm session sharing formats:
  - `m.room_key` -> session sharing format
  - `m.forwarded_room_key` -> session export format
  - Server-side Megolm session backups -> session export format
  - Megolm session file exports -> session export format

- Remove some points of ambiguity regarding terminology.
- Adding links to other parts of the spec.
- Stylistic fixes (rewording for consistency, capitalization, etc).
- Fix for #1080.

Preview: https://pr1088--matrix-spec-previews.netlify.app

### CODE REVIEW

File: content/client-server-api/modules/end_to_end_encryption.md [hunk 1]
```diff
@@ -1257,10 +1257,10 @@ the following format:
 
 `AuthData`
 
-| Parameter  | Type       | Description                                                                                      |
-| -----------| -----------|--------------------------------------------------------------------------------------------------|
-| public_key | string     | **Required.** The curve25519 public key used to encrypt the backups, encoded in unpadded base64. |
-| signatures | Signatures | Optional. Signatures of the ``auth_data``, as Signed JSON                                        |
+| Parameter  | Type       | Description                                                                                                           |
```
Comment (turt2live) [inline content/client-server-api/modules/end_to_end_encryption.md:1260]: we have ways of generating these tables which we should start using for the crypto section too. See https://github.com/matrix-org/matrix-spec/pull/1062 for how we described `m.relates_to`, for example.

**Comment by turt2live** [comment]
while in the area, can I suggest a change to fix https://github.com/matrix-org/matrix-spec/issues/1080 ? 😇

File: content/client-server-api/modules/end_to_end_encryption.md (hunk 1 above)
Comment (dkasak) [reply content/client-server-api/modules/end_to_end_encryption.md:1260]: Unfortunately, I don't have time to figure this one out right now. Mind if we postpone it to a later PR?

Comment (turt2live) [reply content/client-server-api/modules/end_to_end_encryption.md:1260]: Preferably not, as otherwise it won't ever get fixed.

Comment (dkasak) [reply content/client-server-api/modules/end_to_end_encryption.md:1260]: I'd prefer to convert the tables too, but really can't afford to fiddle with this now. But the changes are useful on their own and unrelated to the table generation method, no?

Comment (turt2live) [reply content/client-server-api/modules/end_to_end_encryption.md:1260]: The changes might be useful, but we really can't afford to keep putting off the tech debt.

Comment (dkasak) [reply content/client-server-api/modules/end_to_end_encryption.md:1260]: Yeah, if you do get around to it before me, feel free to do it. Thanks!

**Review by richvdh — COMMENTED** [review]
looks like a significant improvement - a few nits to clean up though

File: content/client-server-api/modules/end_to_end_encryption.md (hunk 1 above)
Comment (richvdh) [reply content/client-server-api/modules/end_to_end_encryption.md:1260]: ok, I have made these changes in #1166. Unfortunately it is going to introduce conflicts with this PR which will now need to be resolved.

**Comment by anoadragon453** [comment]
Hi @dkasak, do you have time to finish this up? The changes look valuable!

**Comment by dkasak** [comment]
Hi, yes, haven't forgotten about it, just the disclosure was keeping me busy. Will be coming back to this shortly.

**Comment by richvdh** [comment]
I'm going to go ahead and close this for now. @dkasak, if you have time to return to it in the future, that would be very much appreciated and we'll be delighted to reopen it!

**Pull request closed without merging**